yarn-project/simulator/src/public/executor.ts

```diff
@@ -169,4 +173,95 @@ export class PublicExecutor {
     const newWorldState = context.persistableState.flush();
     return temporaryConvertAvmResults(execution, newWorldState, result);
   }
+
+  /**
+   * These functions are currently housed in the temporary executor as it relies on access to
+   * oracles like the contractsDB and this is the least intrusive way to achieve this.
+   * When we remove this executor(tracking issue #4792) and have an interface that is compatible with the kernel circuits,
+   * this will be moved to sequencer-client/prover.
+   */
+
+  /**
+   * Generates a proof for an associated avm execution. This is currently only used for testing purposes,
+   * as proof generation is not fully complete in the AVM yet.
+   * @param execution - The execution to run.
+   * @returns An AVM proof and the verification key.
+   */
+  public async getAvmProof(avmExecution: PublicExecution): Promise<Buffer[]> {
+    // The paths for the barretenberg binary and the write path are hardcoded for now.
+    // We additionally need the path to a valid crs for proof generation.
+    const bbPath = '../../barretenberg/cpp';
+    const writePath = '/tmp';
```

Comment:
This might be global on the mainframe, might be worth making this be relative ./tmp, then adding that to gitignore